packages/block-library/src/navigation/edit/responsive-wrapper.js

```diff
@@ -39,12 +39,20 @@ export default function ResponsiveWrapper( {
 
 	const modalId = `${ id }-modal`;
 
+	const dialogProps = {
+		className: 'wp-block-navigation__responsive-dialog',
+		...( isOpen && {
+			role: 'dialog',
+			'aria-modal': true,
+			'aria-label': __( 'Menu' ),
+		} ),
+	};
+
 	return (
 		<>
 			{ ! isOpen && (
 				<Button
 					aria-haspopup="true"
```

Comment:
It's definitely a dialog, so I don't know whether it should be `aria-haspop="dialog"`. It is a menu in a dialog, so maybe that makes it ok to use true. 🤔